hs/src/Reach/Eval/Core.hs

```diff
@@ -310,15 +321,45 @@ env_merge_ imode left righte = foldlM (env_insertp_ imode) left $ M.toList right
 env_merge :: HasCallStack => SLEnv -> SLEnv -> App SLEnv
 env_merge = env_merge_ DisallowShadowing
 
+-- Utilities to track variable
+
+useStrict :: App Bool
+useStrict = liftIO . readIORef =<< asks (sco_use_strict . e_sco)
+
+whenUsingStrict :: App () -> App ()
+whenUsingStrict e = useStrict >>= flip when e
+
+shouldNotTrackVariable :: (SrcLoc, SLVar) -> Bool
+shouldNotTrackVariable (SrcLoc _ _ (Just ReachStdLib), _) = True
+shouldNotTrackVariable (_, "main") = True
+shouldNotTrackVariable (_, "_") = True
+shouldNotTrackVariable _ = False
+
+trackVariable :: (SrcLoc, SLVar) -> App ()
+trackVariable el =
+  whenUsingStrict $ do
+    unused_vars <- asks e_unused_variables
+    unless (shouldNotTrackVariable el) $
+      liftIO $ modifyIORef unused_vars (el :)
+
+markVarUsed :: (SrcLoc, SLVar) -> App ()
+markVarUsed v =
+  whenUsingStrict $ do
```

Comment:
I think this should be removed... we can always mark them used... it's generating the warning that we don't care about

Reply:
In doing so, I believe this will cause inner scopes to not behave as I intended. 

```javascript
A.only(() => {
  'use strict'
  const x = 5;
})
A.publish(x);
```
I would expect `x` to be unused within the scope that `use strict` was declared in. This change will mark `x` as seen although I wanted to `use strict` for only inside the thunk.

Reply:
Yea, I think we disagree about what it means

Reply:
Removed the `whenUsingStrict` check on `markVarUsed` so this now behaves as you intended